src/dataServices/dynamoDbDataService.ts

```diff
@@ -139,19 +151,105 @@ export class DynamoDbDataService implements Persistence {
         };
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     async initiateExport(initiateExportRequest: InitiateExportRequest): Promise<string> {
-        throw new Error('method not implemented');
+        await this.throttleExportRequestsIfNeeded(initiateExportRequest.requesterUserId);
+        // Create new export job
+        const jobId = uuidv4();
+
+        // TODO: Start Export Job Step Function
+        // const stepFunctionArn = await StartStepFunctionAndGetStepFunctionArn
+
+        const params = DynamoDbParamBuilder.buildPutCreateExportRequest(jobId, initiateExportRequest, '');
+        await this.dynamoDb.putItem(params).promise();
+        return jobId;
+    }
+
+    async throttleExportRequestsIfNeeded(requesterUserId: string) {
+        const jobStatuses: ExportJobStatus[] = ['canceling', 'in-progress'];
+        const exportJobItems = await this.getJobsWithExportStatuses(jobStatuses);
+
+        if (exportJobItems) {
+            const numberOfConcurrentUserRequest = exportJobItems.filter(item => {
+                return DynamoDBConverter.unmarshall(item).jobOwnerId === requesterUserId;
+            }).length;
+            if (
+                numberOfConcurrentUserRequest >= this.MAXIMUM_CONCURRENT_REQUEST_PER_USER ||
+                exportJobItems.length >= this.MAXIMUM_SYSTEM_LEVEL_CONCURRENT_REQUESTS
+            ) {
+                throw new TooManyConcurrentExportRequestsError();
+            }
+        }
+    }
+
+    async getJobsWithExportStatuses(jobStatuses: ExportJobStatus[]): Promise<ItemList> {
+        const jobStatusPromises = jobStatuses.map((jobStatus: ExportJobStatus) => {
+            const queryJobStatusParam = DynamoDbParamBuilder.buildQueryExportRequestJobStatus(jobStatus);
+            return this.dynamoDb.query(queryJobStatusParam).promise();

```

Comment:
Doing two queries to get jobs in the state `canceling` and `in-progress` because query only support equality check on partition key.
 https://stackoverflow.com/questions/40316502/aws-dynamodb-querying-all-hashes-in-array

I looked at `BatchGetItems` but that API doesn't support indexes.
https://stackoverflow.com/questions/42501001/how-to-batchget-index-table-in-dynamodb